File: codes/jiheon/53_maximumsubarray.rs
use std::cmp::max;
impl Solution {
	pub fn max_sub_array(nums: Vec<i32>) -> i32{
	 nums.iter().skip(1)
		 .fold((nums[0], nums[0]), |mut acc, num| {
			  acc.0 = max(acc.0 + num, *num); //sum
			  acc.1 = max(acc.0, acc.1); //maximum sum
			  acc }
			).1 
  }
}

File: codes/jeongah/53_maximumsubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_sum = 0
        max_sum = -inf

        for num in nums:
            cur_sum += num

            if num > cur_sum:
                cur_sum = num

            max_sum = max(max_sum, cur_sum)
        
        return max_sum